keymanapp/keyman
# feat(android): Add controls for auto-correct

Follows #12442 and addresses much of #12203 for adding "auto-correct" control on the language settings menu.

From the design meeting decisions, we replace the current "Prediction" and "Correction" toggles with a group of 4 radio-buttons. This has options:

![suggestion radio options](https://github.com/user-attachments/assets/1b9b46a9-130e-4d31-8c66-e7743e48ac80)

TODO for follow-on PR: add KeymanWeb API to apply auto-correct control

## User Testing
**Setup** - Install the PR build of Keyman for Android on a device/emulator

* **TEST_SUGGESTIONS_RADIO_BUTTONS** 
1. Launch Keyman for Android.
2. From the "Get Started" menu, set Keyman as the default system keyboard
3. In the Keyman app, verify the default sil_euro_latin keyboard displays suggestions
4. Launch Chrome and select a text area to type with Keyman as the default keyboard
5. Verify the default sil_euro_latin keyboard displays suggestions
6. Return to the Keyman app --> Keyman settings --> Installed languages --> English to bring up the "English Settings" menu
7. Verify the default suggestion radio selection (at the bottom) is "Predictions with corrections"
8. Select "Predictions only"
9. Verify "Predictions only" is the only selected option
10. Select "Disable suggestions"
11. Verify "Disable suggestions" is the only selected option
12. Exit the menu
13. In the Keyman app, verify the suggestion banner is now disabled (replaced with Keyman banner) 
14. In Chrome select a text field to type in
15. Verify the Keyman suggestion banner is disabled (replaced with Keyman banner)
